src/test/ui/consts/miri_unleashed/const_refers_to_static_cross_crate.rs

```diff
@@ -0,0 +1,35 @@
+// compile-flags: -Zunleash-the-miri-inside-of-you
+// aux-build:static_cross_crate.rs
+#![allow(const_err)]
+
+#![feature(exclusive_range_pattern)]
+#![feature(half_open_range_patterns)]
+
+extern crate static_cross_crate;
+
+// Sneaky: reference to a mutable static.
+// Allowing this would be a disaster for pattern matching, we could violate exhaustiveness checking!
+const SLICE_MUT: &[u8; 1] = { //~ ERROR undefined behavior to use this value
+//~| NOTE encountered a reference pointing to a static variable
+//~| NOTE
+    unsafe { &static_cross_crate::ZERO }
```

Comment:
won't this be worked around if you do `&static_cross_crate::ZERO[0]` here, and then pattern matching suddenly sees the mutable memory?

Reply:
Why would that happen? It's still a reference to a `GlobalAlloc::Static`.

I can add that testcase.

Reply:
You are right, that error does not get caught. But I do not understand why. The pointer should be the same, shouldn't it?

Are we somewhere accidentally "leaking" the internal pointer of the static, as opposed to the lazy one?

Reply:
Ah, never mind -- we have to actually use the constant to cause the error. (Not sure why though, we should be evaluating all consts whether they are used or not...)

Reply:
> Are we somewhere accidentally "leaking" the internal pointer of the static, as opposed to the lazy one?

If that case isn't doing it, then using `&&[u8]` for the static and using `&*static_cross_crate::ZERO` will definitely do it, or by using an enum and getting a reference to a field. Any operation that needs to actually read from the static in order to produce the reference will yield the internal `AllocId` instead of the lazy one.

Reply:
> Any operation that needs to actually read from the static in order to produce the reference will yield the internal AllocId instead of the lazy one.

I don't think that's true. The code in `memory.rs` is very careful to never leak the internal `AllocId` back to the interpreter.

Also, any code that actually *reads* from the static will fail the `before_access_static` dynamic check, and this is already covered.